### PR TITLE
[sparse] BCOO.sum_duplicates: fix handling of explicit zeros

### PR DESCRIPTION
**Pull request by jakevdp**
Also patch up some corner cases in `jnp.unique` that the new tests revealed.

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -5618,14 +5618,21 @@ def _unique(ar, axis, return_index=False, return_inverse=False, return_counts=Fa
...
   if fill_value is not None:
     fill_value = asarray(fill_value, dtype=result.dtype)
   if size is not None and fill_value is not None:
-    valid = lax.expand_dims(arange(size) < mask.sum(), tuple(range(1, result.ndim)))
-    result = where(valid, result, fill_value)
+    if result.shape[0]:
+      valid = lax.expand_dims(arange(size) < mask.sum(), tuple(range(1, result.ndim)))
+      result = where(valid, result, fill_value)
+    else:
+      result = empty_like(result, shape=(size, *result.shape[1:])).at[:].set(fill_value)
```
Comment (soraros) [inline jax/_src/numpy/lax_numpy.py:5635]: Wouldn't `full_like` work in this case?

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py]: Good catch - Done!

**Review by froystig — APPROVED** [review]
